src/gax.ts

```diff
@@ -67,27 +67,53 @@ import {toLowerCamelCase} from './util';
 
 /**
  * Per-call configurable settings for retrying upon transient failure.
+ * @implements {RetryOptionsType}
  * @typedef {Object} RetryOptions
- * @property {String[]} retryCodes
+ * @property {String[] | (function)} retryCodesOrShouldRetryFn
  * @property {BackoffSettings} backoffSettings
+ * @property {(function)} getResumptionRequestFn
  */
 export class RetryOptions {
-  retryCodes: number[];
+  retryCodesOrShouldRetryFn: number[] | ((error: any) => boolean);
```

Comment:
Can we have a better type instead of `any`? This goes to our public interface.

Reply:
Yes, definitely.